aws-android-sdk-mobile-client/src/main/java/com/amazonaws/mobile/client/AWSMobileClient.java

```diff
@@ -327,6 +327,7 @@ private AWSMobileClient() {
         federateWithCognitoIdentityLockObject = new Object();
         showSignInWaitLatch = new CountDownLatch(1);
         initLockObject = new Object();
+        mStore = new DummyStore();
```

Comment:
Seems like the root issue here is that `mStore` is being used before it is initialized. Can we either:
1. Initialize `mStore` with its real implementation, _earlier_?
2. Prevent the calls from happening to the `mStore`, _until_ it exists?

Reply:
1. `AWSMobileClientStore` requires context, which is acquired during initialization, so the real implementation cannot exist until after initialization
2. This would have been ideal, but there are a lot of calls to `mStore` throughout the codebase, which makes it unrealistic to address them all

Strangely enough, the customer reporting the issue claimed that mobile client was successfully, so this PR will help us better identify the root cause.